src/Source.jl

```diff
@@ -156,12 +156,8 @@ function Source(;fullpath::Union{AbstractString,IO}="",
         if categorical
             for i = 1:cols
                 T = columntypes[i]
-                if length(levels[i]) / sum(values(levels[i])) < .67 &&
-                        T !== Missing && Missings.T(T) <: WeakRefString
-                    columntypes[i] = CategoricalArrays.catvaluetype(Missings.T(T), UInt32)
-                    if T >: Missing
-                        columntypes[i] = Union{columntypes[i], Missing}
-                    end
+                if length(levels[i]) / sum(values(levels[i])) < .67 && Missings.T(T) <: WeakRefString
```

Comment:
`T !== Missing` check was here to exclude all-missing columns from the conversion to categorical.